WordPress/gutenberg
# Site Editor: remove redundant style-edit route

## What? How and Why?

The styles site editor routes were introduced in this commit: [`df3d341` (#65619)](https://github.com/WordPress/gutenberg/pull/65619/commits/df3d341af05e19a9911189d99e046c867c72185d)

However the route `match` callbacks were the same for styles and edit: both of them matched `params.canvas !== 'edit'`, effectively registered two routes for "view" mode.

Give that the `/wp_global_styles` route does not have any special layout in edit mode (or for edit + mobile) we can safely delete the `styles-edit` route and fall back to [home-edit](https://github.com/WordPress/gutenberg/blob/52d0b99df9cc1e6e0380e11053ea2ed2575492aa/packages/edit-site/src/components/site-editor-routes/home-edit.js#L10).

Props to @tellthemachines for noticing.

## Testing Instructions

There should be no regression in the way the styles `/wp_global_styles` route works in view and edit modes, either in "desktop" view (essentially wide screens) and "mobile" view (narrow screens).

1. In a block theme, visit `/wp-admin/site-editor.php`
2. Click on Styles, then click on the the canvas to enter edit mode.
3. Click on the WP logo to return back to view mode.
4. Narrow the viewport to entire "mobile" view in both view and canvas modes.

https://github.com/user-attachments/assets/e26ab315-2145-49f8-985e-9a199d4b2400
